src/sql_injection/detect_sql_injection.rs

```diff
@@ -4,25 +4,53 @@ use super::tokenize_query::tokenize_query;
 use crate::diff_in_vec_len;
 
 const SPACE_CHAR: char = ' ';
+pub struct SqlInjectionDetectionResult {
+    pub detected: bool,
+    pub reason: DetectionReason,
+}
+pub enum DetectionReason {
+    // not an injection
+    UserInputNotInQuery,
+    CommonSQLString,
+    FailedToTokenizeQuery,
+    UserInputTooSmall,
+    NoChangesFound,
+    // injection
+    TokensHaveDelta,
+    CommentStructureAltered,
```

Comment:
I think it would be nice to return these, as an integer? So that we know the exact problem inside the agent?

Reply:
would require us to modify this in every agent, knowing the injection <-> not injection ranges, right?